src/js/tech/tech.js

```diff
@@ -449,6 +449,42 @@ class Tech extends Component {
            component instanceof Tech ||
            component === Tech;
   }
+
+  /**
+   * Registers a Tech
+   *
+   * @param {String} name Name of the Tech to register
+   * @param {Object} tech The tecj to register
+   * @static
+   * @method registerComponent
+   */
+  static registerTech(name, tech) {
+    if (!Tech.techs_) {
+      Tech.techs_ = {};
+    }
+
+    Tech.techs_[name] = tech;
+    return tech;
+  }
+
+  /**
+   * Gets a component by name
+   *
+   * @param {String} name Name of the component to get
+   * @return {Component}
+   * @static
+   * @method getComponent
+   */
+  static getTech(name) {
+    if (Tech.techs_ && Tech.techs_[name]) {
+      return Tech.techs_[name];
+    }
+
+    if (window && window.videojs && window.videojs[name]) {
```

Comment:
Not 100% sure on that - or that anything needs to be done, given more reflection.

My original thinking was that if someone were to reach this with a value that wasn't a tech, but was available on the `window`, it would suggest they were doing something really weird. Perhaps the `log.warn` is enough to suggest they did something strange/wrong.